client-sdk/ts-web/src/signature.ts

```diff
@@ -0,0 +1,134 @@
+import * as elliptic from 'elliptic';
+
+import * as hash from './hash';
+import * as misc from './misc';
+import * as types from './types';
+
+export function combineChainContext(context: string, chainContext: string) {
+    return `${context} for chain ${chainContext}`;
+}
+
+export async function prepareSignerMessage(context: string, message: Uint8Array) {
+    return await hash.hash(misc.concat(misc.fromString(context), message));
+}
+
+export interface Signer {
```

Comment:
Why do we have a signer without domain separation contexts? We should make sure that domain separation contexts are used everywhere. Just rename `ContextSigner` to `Signer`.

Reply:
blind no-context signing might be an appealing to devs. I've seen another interface design, Rosetta, expose signing this way https://www.rosetta-api.org/docs/construction_api_introduction.html#detached-key-generation-and-transaction-signing. it might be valuable to have this and the BlindSigner helper for the time being that our context signing construction is only one idea in a broader tooling ecosystem.